src/Microsoft.AspNet.Security.OAuthBearer/OAuthBearerAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,117 @@
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Bearer authentication component which is added to an HTTP pipeline. This constructor is not
+        /// called by application code directly, instead it is added by calling the the IAppBuilder UseOAuthBearerAuthentication 
+        /// extension method.
+        /// </summary>
+        public OAuthBearerAuthenticationMiddleware(
+            RequestDelegate next,
+            IServiceProvider services,
+            IDataProtectionProvider dataProtectionProvider,
+            ILoggerFactory loggerFactory,
+            IOptions<OAuthBearerAuthenticationOptions> options,
+            ConfigureOptions<OAuthBearerAuthenticationOptions> configureOptions)
+            : base(next, services, options, configureOptions)
+        {
+            _logger = loggerFactory.Create<OAuthBearerAuthenticationMiddleware>();
+            if (Options.Notifications == null)
+            {
+                Options.Notifications = new OAuthBearerAuthenticationNotifications();
+            }
+
+            if (Options.SecurityTokenValidators == null)
+            {
+                Options.SecurityTokenValidators = new List<ISecurityTokenValidator> { new JwtSecurityTokenHandler() };
+            }
+
+            if (string.IsNullOrWhiteSpace(Options.TokenValidationParameters.ValidAudience) && !string.IsNullOrWhiteSpace(Options.Audience))
+            {
+                Options.TokenValidationParameters.ValidAudience = Options.Audience;
+            }
+
+            if (Options.ConfigurationManager == null)
+            {
+                if (Options.Configuration != null)
+                {
+                    Options.ConfigurationManager = new StaticConfigurationManager<OpenIdConnectConfiguration>(Options.Configuration);
+                }
+                else if (!(string.IsNullOrWhiteSpace(Options.MetadataAddress) && string.IsNullOrWhiteSpace(Options.Authority)))
+                {
+                    if (string.IsNullOrWhiteSpace(Options.MetadataAddress) && !string.IsNullOrWhiteSpace(Options.Authority))
+                    {
+                        Options.MetadataAddress = Options.Authority;
+                        if (!Options.MetadataAddress.EndsWith("/", StringComparison.Ordinal))
+                        {
+                            Options.MetadataAddress += "/";
+                        }
+
+                        Options.MetadataAddress += ".well-known/openid-configuration";
+                    }
+
+                    HttpClient httpClient = new HttpClient(ResolveHttpMessageHandler(Options));
+                    httpClient.Timeout = Options.BackchannelTimeout;
+                    httpClient.MaxResponseContentBufferSize = 1024 * 1024 * 10; // 10 MB
+
+                    Options.ConfigurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(Options.MetadataAddress, httpClient);
```

Comment:
I talked to Brent offline and understand how this works now. Note you have to specify Authority or MetadataAddress to even reach this point, both of which are optional and have no defaults. for the opaque token scenario you'd hook into the TVP or the notifications and just deal with it. We may find ways to clean up this escalation between scenarios, but I think this is good enough for the initial checkin.